src/test/java/org/jitsi/meet/test/base/FailureListener.java

```diff
@@ -170,6 +179,30 @@ public void onTestFailure(ITestResult testResult)
         }
     }
 
+    @Override
+    public void onConfigurationSuccess(ITestResult itr)
+    {}
+
+    @Override
+    public void onConfigurationFailure(ITestResult itr)
```

Comment:
This callback is called for every configuration method that fails including @AfterClass, @AfterSuite etc. Because our intention here is to handle only "setupClass" I wonder if it's possible to check for method name here.

It may turn out that it's better to keep try catch and cleanupClass where it was, because this logic is kind of specific to AbstractBaseTest. It knows that it has participant to be cleaned up. In my opinion it would be more readable to leave it there.

Even if you leave it where it was the new functionality about adding to failed results should work, because it throws the throwable (results in a failure).

Reply:
The problem is gathering data. Currently there is no screenshot or anything if that fail and we catch it in the AbstractBaseTest.

Reply:
I will check whether we can limit it for just BeforeClass

Reply:
It seems it can be easily done https://static.javadoc.io/org.testng/testng/6.8.21/org/testng/ITestNGMethod.html#isBeforeClassConfiguration()
I will test it.